### DISCUSSION TITLE
gsoc third issue related question (https://github.com/checkstyle/checkstyle/issues/11446)

### DISCUSSION THREAD

**Discussion by TheSuperEngineer** [checkstyle/checkstyle#12290] [General]
While working on "Update WriteTagCheckTest to use verifyWithInlineConfigParser https://github.com/checkstyle/checkstyle/issues/11446", I noticed that we have to change the verify() method with verifyWithInlineConfigParser(). Also at some places verify() method has been changed to execute() method by other contributors.

I wanted to know the following:

1. What does the verify() , verifyWithInlineConfigParser() and execute() method do individually.

2. How is the verify() method different from execute() method. (Does the execute method just run the configs on the target files or something else(In this case how can I play with this method to see different outputs to understand what kind of outputs it produces) )

3. How is verifyWithInlineConfigParser() different from execute(). What was the requirement of the update requested in https://github.com/checkstyle/checkstyle/issues/11446.  Why was the previous state of the codebase that is when we were using verify() method undesirable?

Kindly point me to any material to study this part of the codebase.

**Comment by nrmancuso** [comment]
@shubh220922 as one of our previous GSOC projects, we migrated to a model of “behavior driven development” (BDD). `verifyWithInlineConfigParser` is one of the results of this project. A good way to learn more about any part of the codebase is to use `git blame` to find associated issues and PRs, and read through discussions to understand context and details of implementation.

**Reply by TheSuperEngineer** [reply]
Thanks. On it.

**Reply by romani** [reply]
It completely ok to ask questions in issue as comment or in PR. It will be more contextual

**Reply by TheSuperEngineer** [reply]
Noted.
